pysidt/sidt.py

```diff
@@ -1937,4 +1952,18 @@ def _assign_depths(node, depth=0):
 
 
 def is_prepruned_by_uncertainty(node):
-    return node.rule.uncertainty <= min(item.uncertainty for item in node.items)
\ No newline at end of file
+    return node.rule.uncertainty <= min(item.uncertainty for item in node.items)
+
+def get_bounded_fraction(errs, uncs, confidence_level):
+    t = scipy.stats.norm.ppf((1 + confidence_level) / 2)
+    return np.sum(uncs * t >= np.abs(errs)) / len(errs)
+
+def get_calibration_curve(errs, uncs, n=50):
+    confidence_levels = np.linspace(0, 1, n)
+    proportion_correct = [get_bounded_fraction(errs, uncs, confidence_level) for confidence_level in confidence_levels]
+    return confidence_levels, proportion_correct
+
+def objective_function(scaling_factor, errs, uncs, n = 500):
```

Comment:
Can we make "objective_function" either more specific or embed in another function